agent/agent.go

```diff
@@ -145,13 +145,30 @@ LOOP:
 					continue LOOP
 				}
 				tc = a.prepareNextTask(ctx, nextTask, tc)
-				if err := a.fetchProjectConfig(ctx, tc); err != nil {
+				if err = a.fetchProjectConfig(ctx, tc); err != nil {
 					grip.Error(message.WrapError(err, message.Fields{
 						"message": "error fetching project config; will attempt at a later point",
 						"task":    tc.task.ID,
 					}))
 				}
-				a.resetLogging(lgrCtx, tc)
+				err = a.resetLogging(lgrCtx, tc)
+				if err != nil {
+					detail := &apimodels.TaskEndDetail{
+						Status:      evergreen.TaskFailed,
+						Type:        evergreen.CommandTypeSetup,
+						Description: fmt.Sprintf("invalid logger config: %s", err.Error()),
```

Comment:
This is not the only reason resetLogging could return an error. The error here should probably be more general.

Reply:
changed